test/framework/src/main/java/org/elasticsearch/test/readiness/ReadinessClientProbe.java

```diff
@@ -70,11 +71,11 @@ default void tcpReadinessProbeFalse(Integer port) throws Exception {
 
         try (SocketChannel channel = SocketChannel.open(StandardProtocolFamily.INET)) {
             AccessController.doPrivileged((PrivilegedAction<Void>) () -> {
-                String message = expectThrows(IOException.class, () -> {
+                Throwable throwable = expectThrows(IOException.class, () -> {
                     var result = channelConnect(channel, socketAddress);
                     probeLogger.info("No exception on channel connect, connection success [{}]", result);
-                }).getMessage();
-                assertThat(message, containsString("Connection refused"));
+                });
+                assertThat(throwable, instanceOf(ConnectException.class));
```

Comment:
No need to check instanceof here, you can change expectThrows to expect ConnectException (and no need to capture the exception into a local in this case).